selene-vscode/src/extension.ts

```diff
@@ -160,9 +160,28 @@ export async function activate(context: vscode.ExtensionContext) {
         diagnosticsCollection.set(document.uri, diagnostics)
     }
 
-    vscode.workspace.onDidSaveTextDocument(lint)
+    let disposable: vscode.Disposable
+    function listenToChange() {
+        if (disposable) {
+            disposable.dispose()
+        }
+
+        const lintOn = vscode.workspace.getConfiguration("selene").get<string>("run")
+        if (lintOn === "onSave") {
+            disposable = vscode.workspace.onDidSaveTextDocument(lint)
+        } else { // onType
```

Comment:
Use a switch/case and check explicitly, this comment would be unnecessary if the code is checking for it anyway.